internal/js/modules/k6/browser/common/frame.go

```diff
@@ -1915,6 +1915,10 @@ func (f *Frame) WaitForNavigation(opts *FrameWaitForNavigationOptions) (*Respons
 	select {
 	case evt := <-navEvtCh:
 		if e, ok := evt.(*NavigationEvent); ok {
+			if e.err != nil {
```

Comment:
Note that here I'm early returning the error, while in the `frame_manager.go` code I'm just "saving" the error, to return it later. But I'm not 100% about all possible cases here, and I'm wondering if this is just yet another case in Go where on a given error it's enough with early returning the error, or if the response might have something meaningful that could complement the error 🤔

Reply:
If it's a network error then i wouldn't expect a response object, so _I think_ it's ok to not worry about waiting for a response.